logstash-core/src/main/java/org/logstash/config/ir/compiler/JavaFilterDelegatorExt.java

```diff
@@ -107,11 +107,13 @@ protected IRubyObject doFlush(final ThreadContext context, final RubyHash option
 
     @Override
     protected IRubyObject closeImpl(final ThreadContext context) {
+        filter.close();
         return context.nil;
     }
 
     @Override
     protected IRubyObject doCloseImpl(final ThreadContext context) {
+        filter.close();
         return context.nil;
     }
```

Comment:
:thinking: per my understanding of the code this `doCloseImpl` will _never_ get called, as the `do_*` methods are an implementation dertail of the _ruby_ plugin API's default `close` method wrapping a user-overridable `do_close`.

I am not opposed to merging as-is (I believe it poses zero risk), but I believe we need to chase down exactly what this filter delagator needs in a separate effort, and to trim out the bits that were added unnecessarily.

Reply:
yes `doClose` methods invokes the `close` method, which is expected to be hook implemented into the Plugin's subclasses. 
So `doCloseImpl` keeps the same logic, invokes the `close` method.